src/main/java/com/grapeshot/halfnes/ui/SwingUI.java

```diff
@@ -436,9 +443,6 @@ public final synchronized void render() {
             DisplayMode dm = gd.getDisplayMode();
             int scrnheight = dm.getHeight();
             int scrnwidth = dm.getWidth();
-            //don't ask why this needs to be done every frame,
-            //but it does b/c the canvas keeps resizing itself
-            canvas.setSize(scrnwidth, scrnheight);
```

Comment:
Tested on Windows 7. The issue from the comment didn't appear.

Reply:
Tested OK for me with or without on Windows. I think this was a fix for a Mac bug in 2011 (v034 changelog says "Fixed full screen code for people who still had flicker at bottom") but the SVN history and Google Code issue tracker are long gone.

Reply:
I checked on Mac as well with JDK8. Works well with no issues.